src/common/apple/purchase.ts

```diff
@@ -0,0 +1,247 @@
+  transaction: UserTransaction;
+  user: Pick<User, 'id' | 'subscriptionFlags' | 'coresRole'>;
+  currencyInUSD: number;
+}) => {
+  if (isTest) {
+    return;
+  }
+
+  const blocks: (KnownBlock | Block)[] = [
+    {
+      type: 'header',
+      text: {
+        type: 'plain_text',
+        text: 'Cores purchased :cores: :apple-ico:',
+        emoji: true,
+      },
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Transaction ID:*', data.appTransactionId),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*App Account Token:*',
+            data.appAccountToken,
+          ),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Product:*', data.productId),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Cores:*', transaction.value.toString()),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Purchased by:*',
+            `<https://app.daily.dev/${user.id}|${user.id}>`,
+          ),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Cost:*',
+            new Intl.NumberFormat('en-US', {
+              style: 'currency',
+              currency: 'USD',
+            }).format(currencyInUSD || 0),
+          ),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Currency:*', 'USD'),
+        },
+      ],
+    },
+    {
+      type: 'section',
+      fields: [
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline(
+            '*Cost (local):*',
+            new Intl.NumberFormat('en-US', {
+              style: 'currency',
+              currency: data.currency,
+            }).format((data.price || 0) / 1000),
+          ),
+        },
+        {
+          type: 'mrkdwn',
+          text: concatTextToNewline('*Currency (local):*', data.currency),
+        },
+      ],
+    },
+  ];
+
+  await webhooks.transactions.send({ blocks });
+};
+
+export const handleCoresPurchase = async ({
+  transactionInfo,
+  user,
+}: {
+  transactionInfo: JWSTransactionDecodedPayload;
+  user: Pick<User, 'id' | 'subscriptionFlags' | 'coresRole'>;
+  environment: Environment;
+  notification: ResponseBodyV2DecodedPayload;
+}): Promise<UserTransaction> => {
+  if (!transactionInfo.transactionId) {
+    throw new Error('Missing transactionId in transactionInfo');
+  }
+
+  if (!transactionInfo.productId) {
+    throw new Error('Missing productId in transactionInfo');
+  }
+
+  if (
+    checkUserCoresAccess({
+      user,
+      requiredRole: CoresRole.User,
+    }) === false
+  ) {
+    throw new Error('User does not have access to cores purchase');
```

Comment:
What are we going to do if the user paid and isn't eligible?

Reply:
they are blocked on the client side as well, so its mostly another check, we do it on paddle side as well

we would handle it case by case, do you want to handle somehow differently?

Reply:
but there's also much logic even here that can be reused, am i missing anything?
if we define a common type with all the info from apple/paddle and have one function that checks access, create transactions, etc. it should be better, isn't it? again hard to review this specifically as it's going to change once we go production

Reply:
Not really, transaction creation is different on apple and paddle, also detection of product type. I imagine that any new provider will also have it different, so I think its actually nicer for now to keep the logics separate. 

What could be done is to create a generic provider interface which would standardize what we expect for each provider, but that requires refactoring how apple and paddle webhooks work now (which is quite different) so it would be separate work from this PR.

Reply:
I'll try to add the spec for this in a task, and link back so you can see.

Reply:
Ok makes sense